lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -171,6 +273,198 @@ createSpecializedType(CanSILFunctionType SubstFTy, SILModule &M) const {
                          M.getASTContext());
 }
 
+std::pair<GenericEnvironment *, GenericSignature *>
```

Comment:
At some point you had a utility function that fed a generic signature back through an archetype builder to get a 'more canonical' signature. This was a bug since it should already be canonical the first time. But I see this function does something else, so never mind!